buildLibraries.sh

```diff
@@ -1,141 +1,53 @@
+LIBARCH=false
+HDF5=false
+
+for ARG in "$@"; do
+    if [[ $ARG == "boinc" ]]; then
+        BOINC=true
+        LIBARCH=true
+        echo "Boinc depends on libarchive, libarchive enabled as well."
+    elif [[ $ARG == "libarchive" ]]; then
+        LIBARCH=true
+    elif [[ $ARG == "hdf5" ]]; then
+        HDF5=true
+    else
+        echo "Unknown library $ARG requested."
+        exit 1
     fi
+    echo "Will build $ARG"
+    ALL=false
+done
 
+if [ $BOINC = true ] || [ $ALL = true ]; then
+    git submodule init lib/boinc
+    git submodule update lib/boinc
+    cd lib
+    ./buildBoinc.sh
     cd ..
 fi
 
-if [[ $(pwd) == /afs/* ]]; then
-    #AFS doesn't like hardlinks between files in different directories and configure doesn't check for this corner case...
-    sed -i 's/\/bin\/ln/cp/g' Makefile
-    sed -i 's/\/bin\/ln/cp/g' api/Makefile
-    sed -i 's/\/bin\/ln/cp/g' lib/Makefile
-    #AFS doesn't like parallel make
-    make
-else
-    # Machines with low memory doesn't like an automatic -j
-    make -j4
-fi
-
-#Need to build the boinc/api/boinc_api_fortran.o separately
-cd api
-if [[ -e boinc_api_fortran.o ]]; then
-    #In case we already have such an .o file, it may be of the wrong arch, so recompile.
-    rm boinc_api_fortran.o
-fi
-make boinc_api_fortran.o
-cd ..
-
-cd ..
-
-echo
-echo "####################################"
-echo "### libArchive #####################"
-echo "####################################"
-echo
-echo "**** ZLIB ****"
-echo
-#Support library zlib
-cd zlib
-ZLIB_BASE=$(pwd)
-if [[ -d build ]]; then
-   rm -rf build
-fi
-mkdir build
-if [[ -d install ]]; then
-   rm -rf install
-fi
-mkdir install
-
-cd build
-cmake .. -DCMAKE_INSTALL_PREFIX=$ZLIB_BASE/install -DCMAKE_C_FLAGS=-fPIC -G "Unix Makefiles"
-make
-make install
-cd ..
-
-if [[ $(uname) == MINGW* ]]; then
-    #Windows build creates a differently-named libz.a.
-    cd install/lib
-    cp libzlibstatic.a libz.a
-    cd ../..
-fi
-ZLIB_PATH=$ZLIB_BASE/install/lib/libz.a
-
-cd ..
-
-echo
-echo "**** libArchive ****"
-echo
-
-#Then: LibArchive itself
-if [[ -d libarchive_build ]]; then
-    rm -rf libarchive_build
+if [ $LIBARCH = true ] || [ $ALL = true ]; then
+    git submodule init lib/libarchive
+    git submodule update lib/libarchive
+    cd lib
+    ./buildLibarchive.sh
+    cd ..
 fi
-mkdir libarchive_build
-cd libarchive_build
 
-cmake -DCMAKE_BUILD_TYPE=Release -DENABLE_BZip2=OFF -DENABLE_ZLIB=ON -DENABLE_CAT=OFF -DENABLE_CPIO=OFF -DENABLE_EXPAT=OFF -DENABLE_INSTALL=OFF -DENABLE_LIBXML2=OFF -DENABLE_LZMA=OFF -DENABLE_NETTLE=OFF -DENABLE_OPENSSL=OFF -DENABLE_TAR=OFF -DENABLE_CNG=OFF -DENABLE_ICONV=OFF -DENABLE_TEST=OFF -DZLIB_LIBRARY=$ZLIB_PATH -DZLIB_INCLUDE_DIR=$ZLIB_BASE/install/include -G "Unix Makefiles" ../libarchive -LH
-
-if [[ $(pwd) == /afs/* ]]; then
-    #AFS doesn't like parallel make
-    make
-else
-    # Machines with low memory doesn't like an automatic -j
-    make -j 4
+if [ $HDF5 = true ] || [ $ALL = true ]; then
```

Comment:
Zlib is also a dependency of HDF5 as well. I propose to make a separate script for building zlib, and calling this when HDF or libarchive is true, before building either of them.

Reply:
This has already been done in a different branch I have. There is no rush to merge this, so we can just merge both when the other is ready.

Reply:
If HDF5 static is possible I would also include the feature by default

Reply:
That would trigger the need to ensure it works on all platforms and build types. I would not recommend that at this stage.

Reply:
Agreed. HDF5 is a bit fickle sometimes, and it is not possible to build to a true static binary as it explicitly links dlopen. AFAIK LibArchive has a similar issue.